crates/subspace-farmer/src/cluster/controller/farms.rs

```diff
@@ -45,6 +55,17 @@ struct KnownFarmer {
     last_identification: Instant,
     known_farms: HashMap<FarmIndex, FarmId>,
     close_sender: broadcast::Sender<()>,
+
+    waiting_for_cleanup: bool,
+}
+
+impl KnownFarmer {
+    fn notify_cleanup(&mut self) {
+        if !self.waiting_for_cleanup {
+            self.waiting_for_cleanup = true;
+            let _ = self.close_sender.send(());
+        }
```

Comment:
No need for a separate field, just wrap it into `Option<>` if it needs to happen exactly once, then:
```suggestion
        if let Some(close_sender) = self.close_sender.take() {
            let _ = close_sender.send(());
        }
```